web/models.py

```diff
@@ -163,25 +169,30 @@ def is_healthy(self):
         if not is_healthy:
             self._model_instance.transition_to_state(ModelInstanceStates.FAILED)
         return is_healthy
+    
+    def is_timed_out(self, timeout):
+        last_event_datetime = self._model_instance.updated_at
+        last_generation = self._model_instance.last_generation()
+        if last_generation:
+            last_event_datetime = last_generation.created_at
```

Comment:
Shouldn't this be:
```suggestion
            last_event_datetime = last_generation.updated_at
```
to reflect the time when the generation was completed?

Reply:
I'm pretty sure these values will be the same and so it doesn't matter?

Reply:
Ohh, I thought `created_at` is creation time of generation request object and `updated_at` is when the `generation` attribute is assigned, making them different based on generation time.